### ISSUE TITLE
Excute yosys returns an error message

### ISSUE THREAD

**Issue by Bob0111** [The-OpenROAD-Project/OpenROAD-flow-scripts#2035]
### Subject

[Flow] for any util, flow Makefile, or flow script issues.

### Describe the bug

I install openroad form pre-built binaries, and set the environment path as mentioned in https://openroad-flow-scripts.readthedocs.io/en/latest/user/BuildWithPrebuilt.html. But when I do a make in flow directory, it gives me the logs with errors:

mkdir -p ./results/nangate45/gcd/base ./logs/nangate45/gcd/base ./reports/nangate45/gcd/base ./objects/nangate45/gcd/base
(/usr/bin/time -f 'Elapsed time: %E[h:]min:sec. CPU time: user %U sys %S (%P). Peak memory: %MKB.' /usr/bin/yosys -v 3 -c /home/lenovo/git-project/OpenROAD-flow-scripts/flow/scripts/synth.tcl) 2>&1 | tee ./logs/nangate45/gcd/base/1_1_yosys.log
1. Executing Verilog-2005 frontend: ./designs/src/gcd/gcd.v
2. Executing Liberty frontend: ./objects/nangate45/gcd/base/lib/NangateOpenCellLibrary_typical.lib
3. Executing Verilog-2005 frontend: /home/lenovo/git-project/OpenROAD-flow-scripts/flow/platforms/nangate45/cells_clkgate.v
Using ABC area script.
[FLOW] Extracting clock period from SDC file: ./results/nangate45/gcd/base/clock_period.txt
[FLOW] Setting clock period to 0.46
ERROR: TCL interpreter returned an error: Yosys command produced an error
Command exited with non-zero status 1
Elapsed time: 0:00.11[h:]min:sec. CPU time: user 0.09 sys 0.01 (99%). Peak memory: 29056KB.
make[1]: *** [Makefile:501: do-yosys] Error 1
make: *** [Makefile:504: results/nangate45/gcd/base/1_1_yosys.v] Error 2

I have tried to change the design config file but it didn't work. From the above log, it seems like the bug is related to yosys. 

### Expected Behavior

Output logs without errors and generate the gds file successfully.

### Environment

```shell
[WARNING] Your current OpenROAD version is outdated.
It is recommened to pull the latest changes.
If problem persists, file a github issue with the re-producible test case.
kernel: Linux 6.7.12-amd64
os: Debian GNU/Linux 
cmake version 3.29.3
CMake Error: The source directory "/home/lenovo/git-project/OpenROAD-flow-scripts" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.

I use the newest pre-built OpenRoad binaries, so ignore the cmake error.
```

### To Reproduce

Run make command in the flow directory. 

### Relevant log output

```shell
mkdir -p ./results/nangate45/gcd/base ./logs/nangate45/gcd/base ./reports/nangate45/gcd/base ./objects/nangate45/gcd/base
(/usr/bin/time -f 'Elapsed time: %E[h:]min:sec. CPU time: user %U sys %S (%P). Peak memory: %MKB.' /usr/bin/yosys -v 3 -c /home/lenovo/git-project/OpenROAD-flow-scripts/flow/scripts/synth.tcl) 2>&1 | tee ./logs/nangate45/gcd/base/1_1_yosys.log
1. Executing Verilog-2005 frontend: ./designs/src/gcd/gcd.v
2. Executing Liberty frontend: ./objects/nangate45/gcd/base/lib/NangateOpenCellLibrary_typical.lib
3. Executing Verilog-2005 frontend: /home/lenovo/git-project/OpenROAD-flow-scripts/flow/platforms/nangate45/cells_clkgate.v
Using ABC area script.
[FLOW] Extracting clock period from SDC file: ./results/nangate45/gcd/base/clock_period.txt
[FLOW] Setting clock period to 0.46
ERROR: TCL interpreter returned an error: Yosys command produced an error
Command exited with non-zero status 1
Elapsed time: 0:00.11[h:]min:sec. CPU time: user 0.09 sys 0.01 (99%). Peak memory: 29056KB.
make[1]: *** [Makefile:501: do-yosys] Error 1
make: *** [Makefile:504: results/nangate45/gcd/base/1_1_yosys.v] Error 2
```

**Comment by vijayank88** [comment]
`yosys -V` just check yosys accessible and which version you're using?

**Comment by Bob0111** [comment]
Here is the version information of yosys and klayout:

Yosys 0.33 (git sha1 2584903a060)
KLayout 0.29.1

According to the document, I need yosys newer than 0.21 and klayout newer than 0.28.8. Both are satisfied.

**Comment by rovinski** [comment]
@Bob0111 The document is out of date. I believe right now Yosys 0.39 is required.

**Comment by Bob0111** [comment]
Thanks for updating the document. With the updated version of yosys, I can run the flow without problem and successfully generate GDS file.

**Comment by dzemildzigal** [comment]
I'm still experiencing a very similar issue but I've got yosys version 0.43 installed. I've installed it locally through the OpenROAD-flow-scripts local install. I've also got it installed globally, also 0.43 version. @vijayank88 any thoughts?

**Comment by vijayank88** [comment]
@dzemildzigal I am not facing such issues with latest ORFS commit.
Can you share me what are the commands executed?
Maybe raise new github issue with log and steps you tried.

**Comment by dzemildzigal** [comment]
Hi @vijayank88 I've also tried it with the latest commit of main of ORFS, the errors persist. 

I've gotten confirmation that the project I'm trying to generate GDS for is experiencing issues with the OpenROAD-sky130 flow (the sv2v part since the sources are SystemVerilog) that have never been sorted out and I'm trying to gather more resources in order to help out there to try to push this through.

To address your actual question, sure, I can share more about the issue I'm experiencing but most of the problems are in this github issue in the X-HEEP repository, [here](https://github.com/esl-epfl/x-heep/issues/571).

I fully understand that OpenROAD may not actually be the actual culprit here, as sv2v is most probably the one to blame. If it happens to be that we've flushed out the flow I'll be sure to link the fix over here just because someone tomorrow might stumble on a similar issue when combining sv2v with OpenROAD.
